homeassistant/components/suez_water/coordinator.py

```diff
@@ -0,0 +1,113 @@
+"""Suez water update coordinator."""
+
+import asyncio
+from datetime import datetime, timedelta
+from typing import Any
+
+from pysuez import SuezClient
+from pysuez.client import PySuezError
+
+from homeassistant.core import _LOGGER, HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+
+class AggregatedSensorData:
+    """Hold suez water aggregated sensor data."""
+
+    def __init__(
+        self,
+        value: float,
+        currentMonth: Any,
+        previousMonth: Any,
+        previousYear: Any,
+        currentYear: Any,
+        history: Any,
+        highestMonthlyConsumption: Any,
+        attribution: str,
+    ) -> None:
+        """Create aggregated sensor data."""
+        self.value = value
+        self.currentMonth = currentMonth
+        self.previousMonth = previousMonth
+        self.previousYear = previousYear
+        self.currentYear = currentYear
+        self.history = history
+        self.highestMonthlyConsumption = highestMonthlyConsumption
+        self.attribution = attribution
+
+
+class SuezWaterCoordinator(DataUpdateCoordinator):
+    """Suez water coordinator."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        client: SuezClient,
+        counter_id: int,
+    ) -> None:
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(hours=12),
+            always_update=True,
+        )
+        self._sync_client = client
+        self._counter_id = counter_id
+        self.aggregated_data: None | AggregatedSensorData = None
```

Comment:
Why do we store this data here and why don't we return this in `_async_update_data`? That sets the `coordinator.data` property which is the state of the coordinator.